src/pint/pintk/plk.py

```diff
@@ -524,31 +524,18 @@ def setChoice(self, xid="mjd", yid="pre-fit"):
             if choice.lower() == yid:
                 self.ybuttons[ii].select()
 
+            model = (
+                self.master.psr.postfit_model
+                if self.master.psr.fitted
+                else self.master.psr.prefit_model
+            )
             if choice == "elongation":
-                if not any(
-                    x.startswith("Astrometry")
-                    for x in self.master.psr.prefit_model.components
-                ):
+                if not any(x.startswith("Astrometry") for x in model.components):
                     self.xbuttons[ii].configure(state="disabled")
-                    if not self.master.psr.fitted:
-                        self.ybuttons[ii].configure(state="disabled")
-                if self.master.psr.fitted and not any(
-                    x.startswith("Astrometry")
-                    for x in self.master.psr.pstfit_model.components
-                ):
                     self.ybuttons[ii].configure(state="disabled")
             if choice == "orbital phase":
-                if not any(
-                    x.startswith("Binary")
-                    for x in self.master.psr.prefit_model.components
-                ):
+                if not any(x.startswith("Binary") for x in model.components):
```

Comment:
Wouldn't it be more robust to have `isinstance(model.components[x], PulsarBinary)`?

Reply:
possibly.  it would involve more importing.  I'll see what it looks like

Reply:
Note also that the timing model has a `.is_binary` parameter. That is IMO the proper way to do that check (and where we should be making smarter code for identifying whether a model is a binary or not).